crates/noirc_frontend/src/hir_def/types.rs

```diff
@@ -1403,6 +1429,11 @@ impl Type {
             }
             Slice(elem) => Slice(Box::new(elem.follow_bindings())),
             String(size) => String(Box::new(size.follow_bindings())),
+            FmtString(size, args) => {
+                let size = Box::new(size.follow_bindings());
+                let args = vecmap(args, |arg| arg.follow_bindings());
```

Comment:
Missing the checks for `unify` and `make_subtype_of` in this file

Reply:
I added the checks. They were also missing for `String` so I included that as well. 

I didn't expect the checks to still work when they were not fully implemented. Without the checks if I wrote:
```
let foo = some foo struct;
let x = 5;
let struct_string = if x == 5 { f"{foo}" } else { f"{x}" };
```
I would still get:
```
error: Expected type fmtstr<5, (fooStruct)>, found type fmtstr<3, (Field)>
   ┌─ /mnt/user-data/maxim/noir/crates/nargo_cli/tests/test_data_ssa_refactor/debug_logs/src/main.nr:39:25
   │
39 │     let struct_string = if x == 5 { f"{foo}" } else { f"{x}" };
   │                         --------------------------------------
   │
   = Expected the types of both if branches to be equal
```
Is it just the derivations on `Type` carrying through for the constant lengths?

Reply:
If there's no existing cases for the types, they fall through to the `(lhs, rhs) => lhs == rhs` case which just checks for basic equality and thus fails if there are any type variables that need to be bound.